ecc/goldilocks/twist.go

```diff
@@ -13,11 +13,37 @@ import (
 type twistCurve struct{}
 
 // Identity returns the identity point.
-func (twistCurve) Identity() *twistPoint {
-	return &twistPoint{
-		y: fp.One(),
-		z: fp.One(),
+func (twistCurve) Identity() *twistPoint { return &twistPoint{y: fp.One(), z: fp.One()} }
+
+// IsOnCurve returns true if the point lies on the curve.
+func (twistCurve) IsOnCurve(P *twistPoint) bool {
+	return isOnCurve(&P.x, &P.y, &P.ta, &P.tb, &P.z, true)
```

Comment:
Not sure of the benefit of refactoring the entire functionality of `IsOnCurve` into a private method?

Reply:
because it is not only used here, but also for Curve datatype